coderd/httpmw/cors.go

```diff
@@ -0,0 +1,20 @@
+package httpmw
+
+import (
+	"github.com/go-chi/cors"
+	"net/http"
+)
+
+func CorsMW(allowAll bool, origins ...string) func(next http.Handler) http.Handler {
```

Comment:
Did we not have a CORS MW before?

Reply:
Nope. We don't need CORs for the primary coderd, since all web requests are on the same domain.

The issue is the latency request checks for external proxies uses absolute urls. The local yarn dev server runs on a different port. So for the yarn dev server to hit the `/latency` endpoint, I need to turn on CORs.

This PR blocks all CORs requests to coderd by default if the flag is not `true`

Reply:
This was requested by @BrunoQuaresma. Without it, it was difficult for him to work on the workspace proxy UI.